be/src/connector_sink/hive_chunk_sink.cpp

```diff
@@ -97,6 +97,9 @@ StatusOr<std::unique_ptr<ConnectorChunkSink>> HiveChunkSinkProvider::create_chun
 
     std::unique_ptr<formats::FileWriterFactory> file_writer_factory;
     if (boost::iequals(ctx->format, formats::PARQUET)) {
+        // ensure hive compatibility since hive 3 and lower version accepts specific encoding
```

Comment:
only active in hive sink

Reply:
Why not control this by session variable?

Reply:
It is hard to control this by one SV since the SV may affect behaviors of all existing sink types. 

A more fine-grained behavior is:
- For hive sink, use legacy encoding by default, just like trino.
- For iceberg sink, do not use the legacy encoding because the spec has already specified logical-physical type mapping. https://iceberg.apache.org/spec/#parquet
- For file sink, add a boolean option `parquet.use_legacy_encoding`. If the user wants to write parquet files which may be read by hive, he/she should turn this option on.